tests/scripts/all.sh

```diff
@@ -189,6 +189,10 @@ pre_initialize_variables () {
     if [ -z "${MAKEFLAGS+set}" ]; then
         export MAKEFLAGS="-j$(all_sh_nproc)"
     fi
+    # if CC is not set, use clang by default (if present) to improve build times
+    if [ -z "${CC+set}" ] && (type clang > /dev/null 2>&1); then
```

Comment:
(very minor, more a question) the construction `[ -z "${FOO+set}" ]` is used to be able to distinguish between "set to anything, including the empty string" vs "unset". In this case "set to empty string" surely can't work for `CC`, so I'm not sure why the simpler `[ -z "$CC" ]` isn't used

Reply:
> I'm not sure why the simpler `[ -z "$CC" ]` isn't used

I think that's because we use `set -u` which would give an error here when `CC` is not set. Overall I think `set -u` is very useful (I always set it in my own scripts ever since I wiped my hard drive with something like `rsync --delete some_source/ $DEST_ACTUALLY_UNSET_DUE_TO_A_TYPO/`) but it has the annoying side effect that when you want to actually check if a variable that might legitimated be unset if set or not, then you have to jumps through hoops like this.

Reply:
Ah, good point - thanks

Reply:
That probably would indeed be simpler, but I won't respin just for this.

Reply:
I confirm that `[ -z "${CC+set}" ]` is an idiomatic way of checking whether a variable is unset when `set -u` might be in effect.